### DISCUSSION TITLE
How to generate a burn proof?

### DISCUSSION THREAD

**Discussion by CodeSandwich** [0xPolygon/matic.js#261] [Q&A]
Hi! I'm trying to generate a burn proof for PoS bridge ERC-20 withdrawal from Mumbai.

Here's my burn TX: https://mumbai.polygonscan.com/tx/0xb580d143c27e98d13a5b54e3f432c4ce5e6e5c72c2ba5733d1b4175cec1da397 It's a smart contract call initiated by an incoming FX portal message, which burns some ERC-20 tokens. I've been trying to generate the burn proof with this code:
```ts
const Matic = require("@maticnetwork/maticjs").default;
const matic = new Matic({
  maticProvider: "https://rpc-mumbai.matic.today",
  parentProvider: "https://goerli.infura.io/v3/75aa7935112647bc8cc49d20beafa189",
});
const exit_manager = matic.withdrawManager.exitManager;
const BURN_HASH = '0x3f8a6e175862601408288ecc5032b8f8b94b187ebc4eb82e744456c9bd3379c6';
const SIG = '0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef';
const proof = await exit_manager.buildPayloadForExitHermoine(BURN_HASH, SIG);
console.log("Burn proof:", proof);
```
Unfortunately I'm getting an error:
```
(node:2404637) UnhandledPromiseRejectionWarning: Error: Node does not contain the key
    at /home/zuczek/node_modules/@maticnetwork/maticjs/dist/webpack:/src/libs/ProofsUtil.ts:424:1
    at /home/zuczek/node_modules/merkle-patricia-tree/baseTrie.js:461:14
    at Object.return (/home/zuczek/node_modules/merkle-patricia-tree/baseTrie.js:485:9)
    at processNode (/home/zuczek/node_modules/merkle-patricia-tree/baseTrie.js:288:30)
    at processNode (/home/zuczek/node_modules/merkle-patricia-tree/baseTrie.js:521:5)
    at /home/zuczek/node_modules/merkle-patricia-tree/baseTrie.js:457:5
    at /home/zuczek/node_modules/merkle-patricia-tree/baseTrie.js:180:7
    at processTicksAndRejections (internal/process/task_queues.js:75:11)
```

I've tried `exit_manager.buildPayloadForExit` and `exit_manager.getExitHash`, but they generate exactly the same error.

**Comment by Dimfred** [comment]
Were you able to resolve your issue? I have the exact same code + error. Tho I am using fxportals.

**Reply by ujjwalgupta94** [reply]
@Dimfred are you using `maticjs-fxportal` ?

**Reply by Dimfred** [reply]
Yes I do. So the interesting thing is I withdrew another ERC721 today, and the second one went through. Tho, I used `withdrawTo` in the fxportal example contract and not `withdraw`, but the popped event should be the same / same signature. 

So idk what happened with the first one, but I guess it's stuck now. I hope this doesn't happen on mainnet.

**Reply by ujjwalgupta94** [reply]
I can not say from here, but might be you did something wrong in first tx. Anyway try with multiple tx and see how it goes. Let me know if there is any issue.

**Reply by Dimfred** [reply]
Can't be that I did something wrong in the first tx, cause I did exactly the same in the second. But yeah as said it worked in the second so I guess it's fine.

**Comment by asselstine** [comment]
Here is an updated script:

https://gist.github.com/asselstine/37429e42fcb9807dbfcde9a6376be5d4
